test/jdk/java/awt/ScrollPane/ScrollPaneLimitation/ScrollPaneLimitation.java

```diff
@@ -0,0 +1,174 @@
+                    frame.setSize(200, 200);
+                    frame.pack();
+                    frame.setLocationRelativeTo(null);
+                    frame.setVisible(true);
+                    pane.doLayout();
+
+                    try {
+                        Thread.sleep(1000);
+                    } catch (InterruptedException e) {
+                        throw new RuntimeException("test was interrupted");
+                    }
+                    p = child.getLocation();
+                    System.out.println("Child's initial location " + p);
+                    System.out.println("Pane's insets " + pane.getInsets());
+                    pane.setScrollPosition(0, SCROLL_POS);
+                    p = pane.getScrollPosition();
+                    System.out.println("Scroll pos = " + p);
+                    if (p.y != SCROLL_POS) {
+                        throw new RuntimeException("wrong scroll position");
+                    }
+                    p = child.getLocation();
+                    System.out.println("Child pos = " + p);
+                    if (p.y != -SCROLL_POS) {
+                        if (child.isLightweight()) {
+                            // If it is lightweight it will always have (0, 0) location.
+                            // Check location of its parent - it is Panel and it should
+                            // be at (inset left, inset top + position)
+                            Container cp = child.getParent();
+                            p = cp.getLocation();
+                            if (p.y != -SCROLL_POS) {
+                                throw new RuntimeException("wrong child location");
+                            }
+                        } else {
+                            throw new RuntimeException("wrong child location");
+                        }
+                    }
+
+                    p = pane.getLocationOnScreen();
+                    Dimension d = pane.getSize();
+                    p.x += d.width / 2;
+                    p.y += d.height / 2;
+                });
+                robot.mouseMove(p.x, p.y);
+                robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+                robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+                robot.waitForIdle();
+                synchronized (lock) {
+                    try {
+                        lock.wait(3000);
+                    } catch (InterruptedException e) {
+                        throw new RuntimeException("test was interrupted");
+                    }
+                }
+                if (!mouseWasPressed) {
+                    throw new RuntimeException("mouse was not pressed");
+                }
+            }
+        } finally {
+            EventQueue.invokeLater(() -> {
+                if (frame != null) {
+                    frame.dispose();
+                }
+            });
+        }
+    }
+
+    class MyPanel extends Component {
```

Comment:
Can be made `static`.
```suggestion
    private static class MyPanel extends Component {
```

Reply:
Done.